src/util-mem.h

```diff
@@ -225,6 +281,44 @@ SC_ATOMIC_EXTERN(unsigned int, engine_stage);
     (void*)ptrmem; \
 })
 
+#ifndef HAVE_STRNDUP
+#define SCStrndup(a, b) ({ \
+    char *ptrmem = NULL; \
+    size_t len = (b); \
```

Comment:
Think it would be good to use _len here. I just fixed a shadow declaration issue here: https://github.com/OISF/suricata/pull/3963/commits/35b88991c309a36bee741ced2f1935fb85d1d0d0

The variable 'len' will be a good candidate for existing in a calling function as well.

Reply:
Understand about the _len versus len distinction.  So I will let you make that small change if QA is acceptable.  If you need more changes from me, just let me know.  I removed all of the Unit Tests that needed to make a MaxMind database call in order to succeed.  That left only the parsing tests that verify the code can parse the rule text correctly.